Add an example for force clean checkout at step level (#2261)
buildkite/docs: pages/agent/v3/securing.md

## Forcing clean checkouts

> By default Buildkite will reuse (after cleaning) a previous checkout. This may not be safe if building commits from untrusted sources (for example, 3rd party pull requests). To force a clean checkout every time, set `BUILDKITE_CLEAN_CHECKOUT=true` in the environment. Here is an example to force a clean checkout at step level:

Reviewer: Style updates:

```suggestion
By default, Buildkite will reuse (after cleaning) a previous checkout. This may be unsafe if building commits from untrusted sources (for example, third-party pull requests). To force a clean checkout every time, set BUILDKITE_CLEAN_CHECKOUT=true in the environment. The following example shows how to enforce a clean checkout at the step level:
```

(I'm not intending to change the meaning)

```yaml
steps:
- label: ":github: Clean Checkout"
  command: echo "clean checkout"
  env:
    BUILDKITE_CLEAN_CHECKOUT: true

> ```

Reviewer: Markdown style: Add a blank line after code blocks.

````suggestion
```

````

> In logs for this step, you will find log group "Cleaning pipeline checkout".

Reviewer: Style nitpicks:

```suggestion
In the logs for this step, you will find a log group called "Cleaning pipeline checkout."
```

Reviewer: Actually, I'm guessing this is meant to be how you confirm the clean happened? If so, perhaps we could say that:

```suggestion
You can check that the clean happened by looking for the "Cleaning pipeline checkout" log group after the step runs.
```

## Forcing clean checkouts

> By default, Buildkite will reuse (after cleaning) a previous checkout. This may be unsafe if building commits from untrusted sources (for example, third-party pull requests). To force a clean checkout every time, set BUILDKITE_CLEAN_CHECKOUT=true in the environment. The following example shows how to enforce a clean checkout at the step level:

Reviewer: Missing inline code formatting:

```suggestion
By default, Buildkite will reuse (after cleaning) a previous checkout. This may be unsafe if building commits from untrusted sources (for example, third-party pull requests). To force a clean checkout every time, set `BUILDKITE_CLEAN_CHECKOUT=true` in the environment. The following example shows how to enforce a clean checkout at the step level:
```

```yaml
steps:

> - label: ":github: Clean Checkout"

Reviewer: @dannymidnight, any idea why the emoji is rendering like this in [the preview](https://2261--bk-docs-preview.netlify.app/docs/agent/v3/securing#forcing-clean-checkouts):
![Screenshot 2023-07-21 at 3 20 20 pm](https://github.com/buildkite/docs/assets/119824349/ec39f38f-0cfe-41d9-875a-d40a1e9f32ed)
